packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -495,6 +497,62 @@ class _AnimatedPositionedState extends AnimatedWidgetBaseState<AnimatedPositione
   }
 }
 
+/// Animated widget that automatically transitions the child's size over a given
+/// duration whenever the given child's size changes.
+class AnimatedSize extends SingleChildRenderObjectWidget {
+  /// Creates a widget that animates the size of its child implicitly.
+  ///
+  /// The [curve] and [duration] arguments must not be null.
+  AnimatedSize({
+    Key key,
+    Widget child,
+    this.alignment: FractionalOffset.center,
+    this.curve: Curves.linear,
+    @required this.duration
+  }) : super(key: key, child: child);
```

Comment:
Please add asserts that curve and duration are non-null.  Can the alignment really be null?